packages/shared/src/components/checklist/ChecklistStep.spec.tsx

```diff
@@ -0,0 +1,178 @@
+    expect(description).not.toBeInTheDocument();
+
+    const customComponent = screen.queryByTestId(
+      'checklist-step-custom-component',
+    );
+    expect(customComponent).not.toBeInTheDocument();
+  });
+
+  it('should render title, description and custom component when open', async () => {
+    renderComponent({
+      step,
+      isOpen: true,
+      isActive: false,
+      onToggle: noop,
+      children: customChildren,
+    });
+
+    const title = await screen.findByText(step.title);
+    expect(title).toBeInTheDocument();
+
+    const description = await screen.findByText(step.description);
+    expect(description).toBeInTheDocument();
+
+    const open = await screen.findByTestId('checklist-step-open');
+    expect(open).toBeInTheDocument();
+
+    const customComponent = await screen.findByTestId(
+      'checklist-step-custom-component',
+    );
+    expect(customComponent).toBeInTheDocument();
+  });
+
+  it('should render active state', async () => {
+    renderComponent({
+      step,
+      isOpen: false,
+      isActive: true,
+      onToggle: noop,
+      children: customChildren,
+    });
+
+    const title = await screen.findByText(step.title);
+    expect(title).toBeInTheDocument();
+
+    const description = screen.queryByText(step.description);
+    expect(description).not.toBeInTheDocument();
+
+    const active = await screen.findByTestId('checklist-step-active');
+    expect(active).toBeInTheDocument();
+
+    const customComponent = screen.queryByTestId(
+      'checklist-step-custom-component',
+    );
+    expect(customComponent).not.toBeInTheDocument();
+  });
+
+  it('should render completed state', async () => {
+    renderComponent({
+      step: {
+        ...step,
+        action: {
+          ...step.action,
+          completedAt: new Date(),
+        },
+      },
+      isOpen: false,
+      isActive: true,
+      onToggle: noop,
+      children: customChildren,
+    });
+
+    const title = await screen.findByText(step.title);
+    expect(title).toBeInTheDocument();
+
+    const description = screen.queryByText(step.description);
+    expect(description).not.toBeInTheDocument();
+
+    const completed = await screen.findByTestId('checklist-step-completed');
+    expect(completed).toBeInTheDocument();
+
+    const customComponent = screen.queryByTestId(
+      'checklist-step-custom-component',
+    );
+    expect(customComponent).not.toBeInTheDocument();
+  });
+
+  it('should open on click', async () => {
+    let isOpen = false;
+
+    const onToggle = jest.fn(() => {
+      isOpen = !isOpen;
+    });
+
+    renderComponent({
+      step,
+      isOpen,
+      isActive: false,
+      onToggle,
+      children: customChildren,
+    });
+
+    const checklistElement = await screen.findByTestId('checklist-step');
+    expect(checklistElement).toBeInTheDocument();
+
+    const closed = await screen.findByTestId('checklist-step-closed');
+    expect(closed).toBeInTheDocument();
```

Comment:
Same here

Reply:
@rebelchris what is the remark here, avoiding `data-testid` again or something else?

Reply:
Yes correct, I see my other comment is gone, but you can use `findByTitle('Close')` to find these buttons.

Reply:
Yeah, I updated one in `ChecklistCard`, but this one does not have the title since `CloseButton` is not used.